src/Datadog.Trace/RuntimeMetrics/PerformanceCountersListener.cs

```diff
@@ -0,0 +1,182 @@
+            _gen2Size.Dispose();
+            _lohSize.Dispose();
+            _contentionCount.Dispose();
+        }
+
+        public void Refresh()
+        {
+            if (!_fullInstanceName)
+            {
+                _instanceName = GetSimpleInstanceName();
+            }
+
+            TryUpdateGauge(MetricsPaths.Gen0HeapSize, _gen0Size);
+            TryUpdateGauge(MetricsPaths.Gen1HeapSize, _gen1Size);
+            TryUpdateGauge(MetricsPaths.Gen2HeapSize, _gen2Size);
+            TryUpdateGauge(MetricsPaths.LohSize, _lohSize);
+
+            TryUpdateCounter(MetricsPaths.ContentionCount, _contentionCount, ref _lastContentionCount);
+
+            var gen0 = GC.CollectionCount(0);
+            var gen1 = GC.CollectionCount(1);
+            var gen2 = GC.CollectionCount(2);
+
+            if (_previousGen0Count != null)
+            {
+                _statsd.Increment(MetricsPaths.Gen0CollectionsCount, gen0 - _previousGen0Count.Value);
+            }
+
+            if (_previousGen1Count != null)
+            {
+                _statsd.Increment(MetricsPaths.Gen1CollectionsCount, gen1 - _previousGen1Count.Value);
+            }
+
+            if (_previousGen2Count != null)
+            {
+                _statsd.Increment(MetricsPaths.Gen2CollectionsCount, gen2 - _previousGen2Count.Value);
+            }
+
+            _previousGen0Count = gen0;
+            _previousGen1Count = gen1;
+            _previousGen2Count = gen2;
+        }
+
+        private void InitializePerformanceCounters(string instanceName)
+        {
+            _gen0Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 0 heap size", instanceName);
+            _gen1Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 1 heap size", instanceName);
+            _gen2Size = new PerformanceCounterWrapper(MemoryCategoryName, "Gen 2 heap size", instanceName);
+            _lohSize = new PerformanceCounterWrapper(MemoryCategoryName, "Large Object Heap size", instanceName);
+            _contentionCount = new PerformanceCounterWrapper(ThreadingCategoryName, "Total # of Contentions", instanceName);
+        }
+
+        private void TryUpdateGauge(string path, PerformanceCounterWrapper counter)
+        {
+            var value = counter.GetValue(_instanceName);
+
+            if (value != null)
+            {
+                _statsd.Gauge(path, value);
+            }
+        }
+
+        private void TryUpdateCounter(string path, PerformanceCounterWrapper counter, ref long? lastValue)
+        {
+            var value = counter.GetValue(_instanceName);
+
+            if (value == null)
+            {
+                return;
+            }
+
+            if (lastValue == null)
+            {
+                lastValue = value;
+                return;
+            }
+
+            _statsd.Counter(path, value - lastValue);
+            lastValue = value;
+        }
+
+        private Tuple<string, bool> GetInstanceName()
```

Comment:
weird you use a Tuple class and not a readonly struct instead.

Reply:
Since it's called only once in the whole application lifetime, I was too lazy to define a struct